src/librustc_lint/types.rs

```diff
@@ -102,10 +158,31 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TypeLimits {
                                         );
                                         return;
                                     }
+
+                                    let par_id = cx.tcx.hir().get_parent_node_by_hir_id(e.hir_id);
+                                    if let Node::Expr(par_e) = cx.tcx.hir().get_by_hir_id(par_id) {
+                                        if let hir::ExprKind::Struct(..) = par_e.node {
+                                            if is_range_literal(cx.sess(), par_e)
+                                                && lint_overflowing_range_endpoint(
```

Comment:
This is really indented (12 indents)...

Reply:
It's true, these lints do not lend themselves to readability...

Maybe we could try splitting some of the logic in separate fns?